packages/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java

```diff
@@ -728,6 +734,17 @@ public void setMyLocationButtonEnabled(boolean myLocationButtonEnabled) {
     }
   }
 
+  @Override
+  public void setZoomControlsEnabled(boolean zoomControlsEnabled) {
+    if (this.zoomControlsEnabled == zoomControlsEnabled) {
+      return;
+    }
+    this.zoomControlsEnabled = zoomControlsEnabled;
+    if (googleMap != null) {
```

Comment:
should we put this check at the very beginning of the method? Something like:
```java
if (googleMap == null) {
   return;
}
```
This way, we don't end up setting the `this.zoomControlsEnabled` while googleMap is null

Reply:
I agree, but please note the change has been done in sync with other methods. I believe the goal was to keep data & state in sync and update the map view only when appropriate?

Reply:
hmm. Interesting. @iskakaushik Do you remember any particular reason of not bailing out early in these methods when googleMap is null?